azavea/raster-foundry
# Feature/lol/error messages

Connects #176 

Adds actual error messages to Layers and Images when parts of the processing pipeline fail. Shows these error messages in the processing modal.

![screenshot from 2015-10-16 10 10 38](https://cloud.githubusercontent.com/assets/903219/10543855/4259075a-73ee-11e5-9392-523014b0e50b.png)


### To test
 * Start the worker (see below)
 * Upload one each of the following types of images:
    * Junk data with a tif extension.
    * Ok tif but with only one band (http://landsat-pds.s3.amazonaws.com/L8/139/045/LC81390452014295LGN00/LC81390452014295LGN00_B1.TIF)
    * Ok three band tif.
 * Watch for error messages.
 * Since we have no processor, after 10 minutes, the job using the good image should timeout with an error message as well.

**Starting the worker**
 ```
vagrant ssh worker
cd /opt/worker
envdir /etc/rf.d/env ./manage.py shell
from apps.workers import process
p = process.QueueProcessor()
p.start_polling()
```
